app/controllers/neighborhoods_controller.rb

```diff
@@ -21,6 +22,22 @@ def create
     end
   end
 
+  def edit
+    @neighborhood = Neighborhood.find(params[:id])
+  end
+
+  def update
+    @neighborhood = Neighborhood.find(params[:id])
+    if @neighborhood.update_attributes(neighborhood_params) &&
+        current_user.id == @neighborhood.user_id
+      redirect_to neighborhood_path(@neighborhood),
+        notice: "Neighborhood Edited!"
+    else
+      flash[:alert] = "Your neighborhood was not saved"
```

Comment:
If they are not the user, it wouldn't be "your neighborhood." Might try to be more neutral here:

```ruby
flash[:alert] = "Something went wrong."
```